server/models/ParticipantsModel/index.js

```diff
@@ -18,30 +17,54 @@ const ParticipantsModel = {
       .aggregate(allParticipantsQuery(user, queryParams))
       .toArray(),
   allForAssessment: async (db, chart, filtersService) => {
-    const { filters, filterQueries } = filtersService
+    const { filterQueries } = filtersService
     const query = {
       assessment: chart.assessment,
-      study: { $in: filters.sites, $nin: STUDIES_TO_OMIT },
+      study: { $in: filtersService.requestedSites, $nin: STUDIES_TO_OMIT },
     }
-
     if (filtersService.allFiltersInactive()) {
       return await db
         .collection(collections.assessmentDayData)
-        .find(query,  { projection: ALL_SUBJECTS_MONGO_PROJECTION })
+        .find(query, { projection: ALL_SUBJECTS_MONGO_PROJECTION })
         .stream()
     }
 
-    const includedParticipantsByFilter = await Promise.all(filterQueries.map((query) => {
-      return db.collection(collections.assessmentDayData).distinct('participant', query)
-    }))
-
-    const includedParticipants = new Set(includedParticipantsByFilter.flat())
+    const groupedParticipants = await Promise.all(
+      filterQueries.map((query) => {
+        return db
+          .collection(collections.assessmentDayData)
+          .distinct('participant', query)
+      })
+    )
 
     return await db
       .collection(collections.assessmentDayData)
-      .find({ ...query, participant: { $in: Array.from(includedParticipants)} })
+      .find({
+        ...query,
+        participant: {
+          $in:
+            groupedParticipants.length > 1
```

Comment:
`groupedParticipants` always has a fixed length, I don't think this ternary is necessary

Reply:
This ternary is what fixes this bug
https://github.com/AMP-SCZ/dpdash/issues/663